src/nrniv/nonlinz.h

```diff
@@ -3,10 +3,9 @@
 
 class NonLinImpRep;
 
-class NonLinImp {
+class NonLinImp final {
```

Comment:
Since you mentioned `final` in Slack, my opinion for the case of those classes is that it's not needed. Those classes don't have any virtual function and they don't get derived from any other class so I feel like the `final` definition is a bit superfluous. My interpretation of https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c139-use-final-on-classes-sparingly also follows the same idea

Reply:
There was virtual members before.
This `final` is here to check that no one is going to inherit from `NonLinImp` as everything is possible in `nrn`.

Reply:
Still, those don't inherit from any other class so I find `final` a bit superfluous. If one wants to inherit from `NonLinImp` can still do it by removing the `final` if they want.

Reply:
yes that's also my opinion.

Reply:
For me `final` meant more something like "this class is derived by some other class but there isn't any other class that's derived from it even if it could".
Anyway, I don't have such strong opinion. We can keep the `final` keyword based on your explanation